tests/RoutesTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace HeadlessLaravel\Formations\Tests;
+
+use HeadlessLaravel\Formations\Tests\Fixtures\PostFormation;
+use HeadlessLaravel\Formations\Tests\Fixtures\TagFormation;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Facades\Route;
+
+class RoutesTest extends TestCase
+{
+    use RefreshDatabase;
+
+    public function test_formation_routes_only()
+    {
+        Route::formation('articles', PostFormation::class)->only('index');
+        Route::formation('articles.tags', TagFormation::class)->only(['index']);
+        Route::formation('articles.tags', TagFormation::class)->pivot()->only(['sync']);
+
+        $routes = Route::getRoutes();
+
+        // preloaded from routes.php
+        $routesFilesRoutes = 35;
```

Comment:
I made the Routes class return the array of routes created

This will let you make counts not including the test/Fixtures/routes.php file

Sorry I forgot to add that to the issue.. clever work around though.
```php
$routes = Route::formation('articles', PostFormation::class)->only('index');
```

Reply:
Can you check the 3rd point I sent on slack?

3. We are calling a create method at https://github.com/headlesslaravel/formations/blob/main/src/Routes.php#L244. is there any reason for that?

Because it only works when we use simple

`Route::formation('articles.tags', TagFormation::class)->except(['index']);`

But when we assign it to any variable,

`$a = Route::formation('articles.tags', TagFormation::class)->except(['index']);`

then the instance is not destroyed, and create method is not called at all, and routes are not registered at all.

Reply:
@mitulgolakiya I do it that way so unlimited methods can be chained onto the Routes class

Good point, feel free to call ->create() within the tests to assign it to a variable

Reply:
@dillingham I think instead of calling a create method from tests, we should come up with some better solution. I will try to come up with a solution tomorrow.

Reply:
@mitulgolakiya its calling "pending object" its pretty popular

https://freek.dev/1545-how-to-avoid-large-function-signatures-by-using-pending-objects

I like the destruct method creating the routes because it keeps it open for more chaining

Reply:
This is only a problem for the test, so lets just call create() instead of rebuilding things